app/assets/javascripts/discourse/components/search-advanced-options.js.es6

```diff
@@ -228,7 +230,7 @@ export default Em.Component.extend({
 
     if (match.length !== 0) {
       const existingInput = _.isArray(tags) ? tags.join(',') : tags;
-      const userInput = match[0].replace(REGEXP_TAGS_PREFIX, '');
+      const userInput = match[0].replace(REGEXP_TAGS_PREFIX, '').replace(REGEXP_TAGS_SUFFIX, '');
```

Comment:
I implemented a REGEXP_TAGS_REPLACE that handles matching the prefix and suffix in one shot.